src/Tools/DynamoInstallDetective/ProductLookUp.cs

```diff
@@ -202,8 +289,9 @@ public IInstalledProduct GetProductFromProductCode(string productCode)
 
         public virtual IEnumerable<string> GetProductNameList()
         {
-            var key = OpenKey(REG_KEY64);
-            return key.GetSubKeyNames().Where(s => s.Contains(ProductLookUpName));
+            return RegUtils.GetInstalledProducts().Select(s => s.Value.Item1).Where(s => {
```

Comment:
Match against the displayName (i.e `s.Value.Item1`)

Reply:
hmm, If I'm first coming to this code I wouldn't understand what item1 was - but I think we can now create named tuple fields?
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples#tuple-field-names

Maybe that or adding a comment?

Reply:
Hmm I did add comments here 
https://github.com/DynamoDS/Dynamo/blob/c1f14a59c18d622c1327c934b33f46d51fa6d081/src/Tools/DynamoInstallDetective/ProductLookUp.cs#L32-L33

Did you mean that I should add comments in the Select() above >?

Could not figure out how to define  a dictionary of named Tuples ....

Reply:
I think like this:
`var dict = new Dictionary<string, (string Foo, bool Bar)>();`

Reply:
nice. Thanks